src/server.rs

```diff
@@ -166,10 +271,12 @@ impl<L, M> Server<L, M>
                  -> Result<JoinHandle<Result<()>>> {
         thread::Builder::new()
             .name(format!("raft::Server({})", id))
-            .spawn(move || Server::run(id, addr, peers, store, state_machine))
+            .spawn(move || {
+                let mut server = try!(Server::create(id, addr, peers, store, state_machine, 1500, 3000, 1000, 129));
+                server.run()
+            })
```

Comment:
This part feels like a wart.  I hoped that `spawn()` could be equivalent to `run()`, but `Server` can't be moved into the new thread.  

It may make sense to remove this method altogether, and let application code handle thread creation (so callers can pass in whatever configuration they desire), but I didn't want to take too much of an axe to the code in my first PR. :)